bgpd/bgp_evpn.c

```diff
@@ -5722,7 +5726,7 @@ void bgp_evpn_init(struct bgp *bgp)
 		(int (*)(void *, void *))evpn_route_target_cmp;
 	bgp->vrf_export_rtl->del = evpn_xxport_delete_ecomm;
 	bgp->l2vnis = list_new();
-	bgp->l2vnis->cmp = (int (*)(void *, void *))vni_hash_cmp;
+	bgp->l2vnis->cmp = vni_list_cmp;
```

Comment:
vni_hash_cmp still has old code which returns only "0" or "1". Does this also needs the fix ?  (You may not be the original author of that code though).

Reply:
hash compares return 0 ( not equal ) 1 equal so it's correct.  The bug here is that you *cannot* use hash compare functions with list compare functions( or vice versa )

Reply:
hash comparators return 1 for equivalence and 0 for nonequivalence

Reply:
Got it.